### PR TITLE
Install Capacitor ASAM Plugin

### PR DESCRIPTION
**Pull request by Cullima**
## Description

- Replaces Cordova guided access plugin with Capacitor based ASAM plugin

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: mock/@capacitor/asam.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { AsamPlugin, AsamResult, AsamStatus } from '../../../../../Swift Training/capacitor-plugin-asam';
```
Comment (matthew2564) [inline mock/@capacitor/asam.ts]: Path looks odd, should this not be ```'@dvsa/capacitor-plugin-asam'```?

File: tsconfig.json [hunk 2]
```diff
@@ -30,7 +30,8 @@
       "@app/*": ["src/app/*"],
       "crypto": ["node_modules/crypto-browserify"],
       "stream": ["node_modules/stream-browserify"],
-      "zlib": ["node_modules/browserify-zlib"]
+      "zlib": ["node_modules/browserify-zlib"],
+      "@asam": ["node_modules/@dvsa/capacitor-plugin-asam"]
```
Comment (matthew2564) [inline tsconfig.json]: Do we think this is necessary / has benefits? Other ```@dvsa/*``` packages are not aliased in such way

Comment (Cullima) [reply tsconfig.json]: this is the way Capacitor recomends injecting plugin mocks https://capacitorjs.com/docs/guides/mocking-plugins#jasmine

File: mock/@capacitor/asam.ts (hunk 1 above)
Comment (Cullima) [reply mock/@capacitor/asam.ts]: you're right have corrcted thanks

File: tsconfig.json (hunk 2 above)
Comment (matthew2564) [reply tsconfig.json]: Maybe i have misinterpreted then as to me there should be no need in this file, but in the tsconfig.spec.json you would have

```"@dvsa/capacitor-plugin-asam": ["mock/@capacitor/asam"]```

This then means that anytime we import this plugin into a file we are referencing it via node_modules i.e. ```@dvsa/capacitor-plugin-asam``` and not the ```@asam``` alias.

**Comment by matthew2564** [comment]
PR to be merged in likely ```release-4.8.0```.  To be tested as part of pilot release before rolling out to wider audience. Date of release TBD.

File: tsconfig.json (hunk 2 above)
Comment (Cullima) [reply tsconfig.json]: updated as discussed

**Review by matthew2564 — APPROVED** [review]

**Comment by matthew2564** [comment]
Closed this and reopened via new PR due to number of changes in codebase.

New PR to release = https://github.com/dvsa/des-mobile-app/pull/1104

**Pull request closed without merging**